designer/report-designer/src/org/pentaho/reporting/designer/core/editor/drilldown/swing/SwingGenericUrlDrillDownUiProfile.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ *  terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ *  Foundation.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License along with this
+ *  program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ *  or from the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Lesser General Public License for more details.
+ *
+ *  Copyright (c) 2006 - 2015 Pentaho Corporation..  All rights reserved.
+ */
+
+package org.pentaho.reporting.designer.core.editor.drilldown.swing;
+
+import org.pentaho.reporting.designer.core.editor.drilldown.DrillDownUi;
+import org.pentaho.reporting.designer.core.editor.drilldown.DrillDownUiProfile;
+import org.pentaho.reporting.engine.classic.extensions.drilldown.DrillDownProfile;
+import org.pentaho.reporting.engine.classic.extensions.drilldown.DrillDownProfileMetaData;
+
+import java.util.Locale;
+
+/**
+ * Implementation of DrillDownUiProfile for Swing version of generic-url-drilldown.xul dialog.
+ *
+ * @author Aleksandr Kozlov
+ */
+public class SwingGenericUrlDrillDownUiProfile implements DrillDownUiProfile {
+
+  /** Drill down meta data. */
+  private DrillDownProfile drillDownProfile;
+
+  /** The name of profile for SwingGenericUrlDrillDownUi. */
+  public static final String[] NAMES = { "generic-url", "local-url" };
```

Comment:
Public static arrays are always a bad thing. Anyone (even from scripts running in a report or kettle-step) can do bad things to the contents of the array. 

Can you rewrite this (and the other occurences) to use a read-only list instead? 
Collections.unmodifiableList(Arrays.asList(....)) should do the trick and no one can ever change the contents of the list.